src/server/bg_services/lifecycle.js

```diff
@@ -48,65 +138,81 @@ async function handle_bucket_rule(system, rule, j, bucket) {
+        dbg.log0('LIFECYCLE PROCESSING rule.expiration.days:', rule.expiration.days);
+        const res = await delete_expired_objects(system, bucket, rule, reply_objects);
+        num_objects_deleted = res.num_objects_deleted;
+        //dbg.log0("LIFECYCLE PROCESSING res =", res);
+        if (res.deleted_objects) {
+            const writes = [];
+            for (const deleted_obj of res.deleted_objects) {
+                for (const notif of bucket.notifications) {
+                    if (check_notif_relevant(notif, {
+                        op_name: 'lifecycle_delete',
+                        s3_event_method: deleted_obj.created_delete_marker ? 'DeleteMarkerCreated' : 'Delete',
+                    })) {
+                        //remember that this deletion needs a notif for this specific notification conf
+                        writes.push({notif, deleted_obj});
+                    }
                 }
             }
-        }
 
-        //if any notifications are needed, write them in notification log file
-        //(otherwise don't do any unnecessary filesystem actions)
-        if (writes.length > 0) {
-            let logger;
-            try {
-                logger = get_notification_logger('SHARED');
-                await P.map_with_concurrency(100, writes, async write => {
-                    const notif = compose_notification_lifecycle(write.deleted_obj, write.notif, bucket);
-                    logger.append(JSON.stringify(notif));
-                });
-            } finally {
-                if (logger) logger.close();
+            //if any notifications are needed, write them in notification log file
+            //(otherwise don't do any unnecessary filesystem actions)
+            if (writes.length > 0) {
+                let logger;
+                try {
+                    logger = get_notification_logger('SHARED');
+                    await P.map_with_concurrency(100, writes, async write => {
+                        const notif = compose_notification_lifecycle(write.deleted_obj, write.notif, bucket);
+                        logger.append(JSON.stringify(notif));
+                    });
+                } finally {
+                    if (logger) logger.close();
+                }
             }
         }
     }
 
+    if (rule.expiration?.expired_object_delete_marker) {
+        await delete_expired_delete_markers(
+            system,
+            bucket.name,
+            rule,
+            reply_objects
+        );
+    }
+
+    // Check if rule.AbortIncompleteMultipartUpload exists - 
+    // if it does, delete incomplete parts if DaysAfterInitiation has passed
+    if (rule.abort_incomplete_multipart_upload?.days_after_initiation) {
+        await delete_incomplete_multipart_uploads(
+            system,
+            bucket.name,
+            rule,
+            reply_objects
+        );
+    }
+
+    if (rule.noncurrent_version_expiration?.noncurrent_days) {
```

Comment:
here  we are checking for the `noncurrent_days` and version delete method(`delete_noncurrent_versions()`) is getting called. Should we add `or` condition for `newer_noncurrent_versions` also